server/middleware/if_param.js

```diff
@@ -0,0 +1,24 @@
+const errors = require('../lib/errors');
+
+/**
+ *  Parameters optionally for a route.
+ */
+module.exports = function(param, value) {
+
+  return function (req, res, next) {
+    // if param doesn't exist, go to next route
+    if (!(req.query[param]) &&
+      !(req.headers[param]) &&
+      !(req.body[param])) {
```

Comment:
I'm not a fan of defining multiple ways to pass parameters to fulfill the same function. Saw it in other places in the code too. Only seems to increase complexity of the code.